internal/util/fs.go

```diff
@@ -0,0 +1,23 @@
+package util
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// EnsureEmptyDirectory ensures the directory given by `path` is empty.
+// If the directory does not exist, it will be created with permission bits
+// given by `perm`.
+func EnsureEmptyDirectory(path string, perm fs.FileMode) error {
+	entries, err := os.ReadDir(path)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	for _, entry := range entries {
+		if err := os.RemoveAll(filepath.Join(path, entry.Name())); err != nil {
+			return err
+		}
+	}
```

Comment:
it should be possible to make it more concise and readable by replacing lines 13-21 with just:
```go
if !os.Stat(path).IsDir() {
   return fmt.Errorf("path %q is not a directory", path) // or similar 
}
err := os.RemovAll(path)
...

Reply:
The problem is, path can point to a pod mount, in which case os.RemoveAll will fail. That's why we step through and delete each of the entries, rather than nuke the whole directory =S

Reply:
thanks for the explanation, maybe a short doc comment for people potentially wanting to refactor it in the future?